php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -50,6 +58,104 @@ public ExpressionTree EXPRESSION() {
    * [ START ] Expression
    */
 
+  public LiteralTree COMMON_SCALAR() {
+    return b.<LiteralTree>nonterminal(PHPLexicalGrammar.COMMON_SCALAR)
+      .is(b.firstOf(
+        f.heredocLiteral(b.token(PHPLexicalGrammar.HEREDOC)),
+        NUMERIC_LITERAL(),
+        // FIXME: when ENCAPS_LIST is migrated
+        //f.stringLiteral(b.token(PHPLexicalGrammar.STRING_LITERAL)),
+        f.booleanLiteral(b.token(PHPLexicalGrammar.BOOLEAN_LITERAL)),
+        f.nullLiteral(b.token(PHPLexicalGrammar.NULL)),
+        f.magicConstantLiteral(b.firstOf(
+          b.token(PHPLexicalGrammar.CLASS_CONSTANT),
+          b.token(PHPLexicalGrammar.FILE_CONSTANT),
+          b.token(PHPLexicalGrammar.DIR_CONSTANT),
+          b.token(PHPLexicalGrammar.FUNCTION_CONSTANT),
+          b.token(PHPLexicalGrammar.LINE_CONSTANT),
+          b.token(PHPLexicalGrammar.METHOD_CONSTANT),
+          b.token(PHPLexicalGrammar.NAMESPACE_CONSTANT),
+          b.token(PHPLexicalGrammar.TRAIT_CONSTANT)))));
+  }
+
+  public LiteralTree NUMERIC_LITERAL() {
+    return b.<LiteralTree>nonterminal()
+      .is(f.numericLiteral(b.token(PHPLexicalGrammar.NUMERIC_LITERAL)));
+  }
+
+  public ExpandableStringLiteralTree EXPANDABLE_STRING_LITERAL() {
+    return b.<ExpandableStringLiteralTree>nonterminal(PHPLexicalGrammar.ENCAPS_STRING_LITERAL)
+      .is(f.expandableStringLiteral(
+        b.token(PHPLexicalGrammar.SPACING),
+        b.token(PHPLexicalGrammar.DOUBLE_QUOTE),
+        b.oneOrMore(
+          b.firstOf(
+            ENCAPSULATED_STRING_VARIABLE(),
+            EXPANDABLE_STRING_CHARACTERS()
+          )
+        ),
+        b.token(PHPLexicalGrammar.DOUBLE_QUOTE)
+      ));
+  }
+
+  public ExpressionTree ENCAPSULATED_STRING_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.ENCAPS_VAR)
+      .is(b.firstOf(
+          ENCAPSULATED_SEMI_COMPLEX_VARIABLE(),
+          ENCAPSULATED_SIMPLE_VARIABLE(),
+          ENCAPSULATED_COMPLEX_VARIABLE()
+        )
+      );
+  }
+
+  public ExpressionTree ENCAPSULATED_COMPLEX_VARIABLE() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)